### PR TITLE
[NEP-3511] Deactivate quick filters based on removed filter

### PR DESCRIPTION
**Pull request by louman**
https://jobready.atlassian.net/browse/NEP-3511

qui

Neptune's part: https://github.com/jobready/neptune/pull/1949

### CODE REVIEW

File: src/stores/FilterBarStore.js [hunk 1]
```diff
@@ -163,6 +164,17 @@ export class FilterBarStore {
     this.emitChange();
   }
 
+  deactivateQuickFiltersBasedOnRemovedFilter(filterName, quickFilters) {
+    var self = this;
+    quickFilters.map(function (quickFilter) {
+      Object.keys(quickFilter.filters).map(function (quickFilterName) {
```
Comment (andrba) [inline src/stores/FilterBarStore.js]: should we create a generator for such loops inside loops? They appear in multiple places now

Comment (louman) [reply src/stores/FilterBarStore.js]: Could you please post an example?

Comment (andrba) [reply src/stores/FilterBarStore.js]:
```js
function *simpleGenerator() {
    for (var i = 0; i < 3; i++) {
    	yield i;
    }
}

for (var v of simpleGenerator()) {
    console.log(v);
}
// 0
// 1
// 2
```

Comment (louman) [reply src/stores/FilterBarStore.js]: @andrba Updated

**Comment by louman** [comment]
@andrba Unless you have anything else to add, could you approved the pr?

**Review by andrba — APPROVED** [review]
👍

**Pull request merged**